### PR TITLE
Improve server side input validation 

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*
#90 

*Description of changes:*
- input validation
  - url: pass relative url instead of full url, compose url on server side wiht `localhost:5601`
  - text fields: validate non-empty.
     - `report Name`: validate allowed characters using regex
     -  `email recipients`: validate email address format
  - timestamp fields: validate positive number
  - cron-expression: validate unix cron-expression (5 digits)
  
  - time duration: validate ISO format for time duration. e.g. `PT10M` (10 mintues)
- Decouple create visual report functions to another file, prepare for project layout re-arrangement. Will implement in future PRs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/public/components/main/main_utils.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export const addReportsTableContent = (data) => {
       //TODO: wrong name
       timeCreated: report.time_created,
       state: report.state,
-      url: report.query_url,
+      url: `${location.host}${report.query_url}`,
```
Comment (davidcui-amzn) [inline kibana-reports/public/components/main/main_utils.tsx:66]: What does `location.host` add?

File: kibana-reports/server/model/index.ts [hunk 2]
```diff
@@ -27,32 +29,56 @@ import {
 } from '../routes/utils/constants';
 
 export const dataReportSchema = schema.object({
-  base_url: schema.uri(),
+  base_url: schema.string({
+    validate(value) {
```
Comment (davidcui-amzn) [inline kibana-reports/server/model/index.ts:33]: If the validation fails will this still return an error to the client-side and trigger the error toasts in the UI?

**Review by davidcui-amzn — APPROVED** [review]
LGTM- just have some minor questions

File: kibana-reports/public/components/main/main_utils.tsx (hunk 1 above)
Comment (zhongnansu) [reply kibana-reports/public/components/main/main_utils.tsx:66]: the `query_url` is now relative, need to add the host as well for the link rendered in report table -> report source field to be accessible.

File: kibana-reports/server/model/index.ts (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/server/model/index.ts:38]: It will, it will also include the error message

**Review by akbhatta — APPROVED** [review]

**Pull request merged**